automation/src/test/scala/org/broadinstitute/dsde/workbench/sam/api/SamApiSpec.scala

```diff
@@ -184,169 +170,20 @@ class SamApiSpec extends AnyFreeSpec with BillingFixtures with Matchers with Sca
-
-          val petSaKeyNew = Sam.user.petServiceAccountKey(projectName)(user.makeAuthToken())
-          val petSaEmailNew = getFieldFromJson(petSaKeyNew, "client_email")
-          val petSaKeyIdNew = getFieldFromJson(petSaKeyNew, "private_key_id")
-
-          register cleanUp Sam.user.deletePetServiceAccountKey(projectName, petSaKeyIdNew)(user.makeAuthToken())
-
-          petSaEmailOriginal should equal(petSaEmailNew) //sanity check to make sure the SA is the same
-          petSaKeyIdOriginal should not equal petSaKeyIdNew //make sure we were able to generate a new key and that a new one was returned
-        }
-      }
-    }
-
-    "should get an access token for a user's own pet service account" in {
-      val user = UserPool.chooseStudent
-
-      withCleanBillingProject(UserPool.chooseProjectOwner, List(user.email)) { project =>
-        // get my pet's email
-        val petEmail1 = Sam.user.petServiceAccountEmail(project)(user.makeAuthToken())
-
-        val scopes = Set("https://www.googleapis.com/auth/userinfo.email", "https://www.googleapis.com/auth/userinfo.profile")
-
-        // get my pet's token
-        val petToken = Sam.user.petServiceAccountToken(project, scopes)(user.makeAuthToken())
-
-        // convert string token to an AuthToken
-        val petAuthToken = new AuthToken {
-          override def buildCredential() = ???
-
-          override lazy val value = petToken
-        }
-
-        // get my pet's email using my pet's token
-        val petEmail2 = Sam.user.petServiceAccountEmail(project)(petAuthToken)
-
-        // result should be the same
-        petEmail2 shouldBe petEmail1
-      }
-    }
-
-    "should arbitrarily choose a project to return a pet key for when the user has existing pets" in {
-      val user = UserPool.chooseStudent
-      val userToken = user.makeAuthToken()
-
-      withCleanBillingProject(UserPool.chooseProjectOwner, List(user.email)) { project1 =>
-        withCleanBillingProject(UserPool.chooseProjectOwner, List(user.email)) { project2 =>
-          // get my pet's email thru the arbitrary key endpoint
-          val petEmailArbitrary = getFieldFromJson(Sam.user.arbitraryPetServiceAccountKey()(userToken), "client_email")
+      val email = WorkbenchEmail(Sam.user.status()(authToken1).get.userInfo.userEmail)
+      val username = email.value.split("@").head
+      val userId = Sam.user.status()(authToken1).get.userInfo.userSubjectId
 
-          //get the user's subject id for comparison below
-          val userSubjectId = Sam.user.status()(userToken).get.userInfo.userSubjectId
+      val petSAEmail = ServiceAccountAuthTokenFromJson(Sam.user.arbitraryPetServiceAccountKey()(authToken1)).buildCredential().getServiceAccountId
 
-          // result should be a pet associated with the user
-          assert(petEmailArbitrary.contains(userSubjectId))
-        }
-      }
-    }
-
-    "should arbitrarily choose a project to return a pet key for when the user has no existing pets" in {
```

Comment:
does not test anything different that `should give pets the same access as their owners` above